tests/integration/kethaa/test_ethaa.py

```diff
@@ -0,0 +1,103 @@
+import os
+import sys
+
+sys.path.append(os.path.join(os.path.dirname(__file__), "../"))
+import pytest
+import web3
+from eth_account._utils.legacy_transactions import (
+    serializable_unsigned_transaction_from_dict,
+)
+from eth_keys import keys
+from starkware.starknet.testing.starknet import Starknet
+from starkware.starkware_utils.error_handling import StarkException
+
+from tests.utils.bits import combine_ints, to_uint
+from tests.utils.signer import BaseSigner, MockEthSigner
+
+txdict = dict(
+    nonce=1,
+    chainId=9001,
+    maxFeePerGas=1000,
+    maxPriorityFeePerGas=667667,
+    gas=999999999,
+    to=bytes.fromhex("95222290dd7278aa3ddd389cc1e1d165cc4bafe5"),
+    value=10000000000000000,
+    data=b"",
+)
+
+
+@pytest.mark.asyncio
+async def test_address_compute(setup_kethaa):
+    starknet = await Starknet.empty()
+    (deployer, account, private_key, evm_address) = setup_kethaa
+
+    call_info = await deployer.compute_starknet_address(evm_address=evm_address).call()
+
+    assert call_info.result.contract_address == account.contract_address
```

Comment:
so the test here tests that the Deployer's API `compute_starknet_address` returns a result consistent with what is done internally when deploying an account. Test should hence be:

```
class TestExternallyOwnedAccount:
    class TestComputeStarknetAddress:
        async def test_should_return_same_address_as_deployed_contract():
            assert call_info.result.contract_address == account.contract_address
```

and there is also another test regarding the `get_eth_address`:

```
class TestExternallyOwnedAccount:
    class TestGetEthAddress:
        async def test_should_return_the_eth_address_used_at_deploy():
            assert await account.get_eth_address() == evm_address
```